src/turbomind/kernels/attention/test_attention.cu

```diff
@@ -298,19 +300,16 @@ int test_attention()
     thrust::universal_vector<int>   cu_seqlens(kBatchSize + 1);
     thrust::universal_vector<int>   cu_kv_lens(kBatchSize + 1);
 
-    thrust::universal_vector<float> partial_M(kTokenNum * kHeadNum * kMaxSplitK);
-    thrust::universal_vector<float> partial_L(kTokenNum * kHeadNum * kMaxSplitK);
-    thrust::universal_vector<float> partial_O(kTokenNum * kHeadNum * kMaxSplitK * kHeadDim);
-    thrust::universal_vector<int>   split_cnt(kTokenNum);
-    thrust::universal_vector<int>   semaphores(kTokenNum * kHeadNum * kMaxSplitK);
-
-    thrust::universal_vector<T> kv_cache_quant_data(kBatchSize * KvHeadNum * 2 * kContextLen * 2);
-    thrust::fill(kv_cache_quant_data.begin(), kv_cache_quant_data.end(), T{0.});
+    thrust::device_vector<float> partial_M(kTokenNum * kHeadNum * kMaxSplitK);
```

Comment:
what's the difference between device_vector and universal_vector?

Reply:
compute-santizer failed to detect OOB access for universial_vector.